### PR TITLE
(exa PR) 1196: Update icons to nerd-font-3, replace few icons, handle compressed files separately

### PR DESCRIPTION
**Pull request by cafkafk**
https://github.com/ogham/exa/pull/1196

### CODE REVIEW

**Comment by sbatial** [comment]
I've looked into this and I am all for it.

I would definitely consider it a breaking change so as always: There will be setups broken.
But that's a step worth taking considering we are taking about a _font_.

I was confused earlier because https://github.com/ogham/exa/pull/1094 showed up as the Gentoo logo in my terminal.

**Comment by sbatial** [comment]
I'd just say we should decide on either blocking this by all other icon changes or vice versa because they have obvious overlap.

**Comment by cafkafk** [comment]
> I'd just say we should decide on either blocking this by all other icon changes or vice versa because they have obvious overlap.

I don't use icons at all, so I'll deffer to your judgment on this.

I think we should mark all issues dealing with icons with a label first and then assess the whole situation, right now I don't really have an overview. Also, if we are lucky and the contributor that made the PR shows up I'd love to hear what they think about it.

**Comment by sbatial** [comment]
Ok, in that case: Let's block this one. It's easier to change/review one big one than every other small icon change in this case.

Marking all the icon-related ones sounds smart. :+1:

**Comment by cafkafk** [comment]
This is affected by https://github.com/cafkafk/eza/pull/28

**Comment by cafkafk** [comment]
I'm gonna try to have a crack at this but it's probably easier as a new pr, the merge would annoying. I'll add @zer0-x as co-author on the commits.

File: src/info/filetype.rs [hunk 1]
```diff
@@ -28,25 +28,24 @@ impl FileExtensions {
             "build.gradle", "pom.xml", "Rakefile", "package.json", "Gruntfile.js",
             "Gruntfile.coffee", "BUILD", "BUILD.bazel", "WORKSPACE", "build.xml", "Podfile",
             "webpack.config.js", "meson.build", "composer.json", "RoboFile.php", "PKGBUILD",
-            "Justfile", "Procfile", "Dockerfile", "Containerfile", "Vagrantfile", "Brewfile",
-            "Gemfile", "Pipfile", "build.sbt", "mix.exs", "bsconfig.json", "tsconfig.json",
+            "Justfile", "justfile", "Procfile", "Dockerfile", "Containerfile", "Vagrantfile",
+            "Brewfile", "Gemfile", "Pipfile", "build.sbt", "mix.exs", "bsconfig.json", "tsconfig.json",
```
Comment (cafkafk) [inline src/info/filetype.rs:32]: done

File: src/info/filetype.rs [hunk 2]
```diff
@@ -28,25 +28,24 @@ impl FileExtensions {
...
         file.extension_is_one_of( &[
-            "png", "jfi", "jfif", "jif", "jpe", "jpeg", "jpg", "gif", "bmp",
-            "tiff", "tif", "ppm", "pgm", "pbm", "pnm", "webp", "raw", "arw",
-            "svg", "stl", "eps", "dvi", "ps", "cbr", "jpf", "cbz", "xpm",
-            "ico", "cr2", "orf", "nef", "heif", "avif", "jxl", "j2k", "jp2",
-            "j2c", "jpx",
+            "png", "jfi", "jfif", "jif", "jpe", "jpeg", "jpg", "gif", "bmp", "tiff",
+            "tif", "ppm", "pgm", "pbm", "pnm", "webp", "raw", "arw", "svg", "stl", "eps",
+            "dvi", "ps", "cbr", "jpf", "cbz", "xpm", "ico", "cr2", "orf", "nef",
+            "heif", "avif", "jxl", "j2k", "jp2", "j2c", "jpx", "pxm",
```
Comment (cafkafk) [inline src/info/filetype.rs:41]: pxm is the only addition here

Comment (cafkafk) [reply src/info/filetype.rs:41]: this is done

File: src/info/filetype.rs [hunk 3]
```diff
@@ -28,25 +28,24 @@ impl FileExtensions {
...
         ])
     }
 
     fn is_video(&self, file: &File<'_>) -> bool {
         file.extension_is_one_of( &[
             "avi", "flv", "m2v", "m4v", "mkv", "mov", "mp4", "mpeg",
-            "mpg", "ogm", "ogv", "vob", "wmv", "webm", "m2ts", "heic",
+            "mpg", "ogm", "ogv", "vob", "wmv", "webm", "m2ts", "heic", "video",
```
Comment (cafkafk) [inline src/info/filetype.rs:48]: I've never seen this video format before, need example

File: src/info/filetype.rs [hunk 4]
```diff
@@ -79,9 +78,10 @@ impl FileExtensions {
 
     fn is_compressed(&self, file: &File<'_>) -> bool {
         file.extension_is_one_of( &[
-            "zip", "tar", "Z", "z", "gz", "bz2", "a", "ar", "7z",
-            "iso", "dmg", "tc", "rar", "par", "tgz", "xz", "txz",
-            "lz", "tlz", "lzma", "deb", "rpm", "zst", "lz4", "cpio",
+            "zip", "tar", "taz", "Z", "z", "gz", "bz", "bz2", "a", "ar",
```
Comment (cafkafk) [inline src/info/filetype.rs:81]: taz and bz added here

File: src/info/filetype.rs [hunk 5]
```diff
@@ -79,9 +78,10 @@ impl FileExtensions {
...
+            "7z", "iso", "dmg", "tc", "rar", "par", "tgz", "xz", "txz",
+            "lz", "tlz", "lzma", "deb", "rpm", "zst", "lz4", "cpio", "lzh",
+            "lzo", "tbz", "tbz2", "tz", "tzo"
```
Comment (cafkafk) [inline src/info/filetype.rs:84]: "lzh",
            "lzo", "tbz", "tbz2", "tz", "tzo"

all seem new

File: src/info/filetype.rs (hunk 4 above)
Comment (cafkafk) [reply src/info/filetype.rs:81]: done

File: src/info/filetype.rs (hunk 5 above)
Comment (cafkafk) [reply src/info/filetype.rs:84]: done

File: src/info/filetype.rs [hunk 6]
```diff
@@ -137,6 +137,9 @@ impl FileIcon for FileExtensions {
         else if self.is_video(file) {
             Some(Icons::Video.value())
         }
+        else if self.is_compressed(file) {
+            Some(Icons::Compressed.value())
+        }
```
Comment (cafkafk) [inline src/info/filetype.rs:142]: done

File: src/output/icons.rs [hunk 7]
```diff
@@ -16,14 +16,16 @@ pub enum Icons {
...
+    Compressed,
 }
 
 impl Icons {
     pub fn value(self) -> char {
         match self {
-            Self::Audio  => '\u{f001}',
-            Self::Image  => '\u{f1c5}',
-            Self::Video  => '\u{f03d}',
+            Self::Audio  => '\u{f001}', // 
```
Comment (cafkafk) [inline src/output/icons.rs:25]: this seems wrong

File: src/output/icons.rs [hunk 8]
```diff
@@ -16,14 +16,16 @@ pub enum Icons {
...
+            Self::Image  => '\u{f1c5}', // 
+            Self::Video  => '\u{f03d}', // 
+            Self::Compressed => '\u{f410}', // 
```
Comment (cafkafk) [inline src/output/icons.rs:28]: done

File: src/output/icons.rs [hunk 9]
```diff
@@ -48,27 +50,23 @@ pub fn iconify_style(style: Style) -> Style {
...
-        m.insert("bin", '\u{e5fc}'); // 
-        m.insert("config", '\u{e5fc}'); // 
+        m.insert("Cargo.lock", '\u{f1617}'); // 󱘗
         m.insert("docker-compose.yml", '\u{f308}'); // 
         m.insert("Dockerfile", '\u{f308}'); // 
         m.insert("ds_store", '\u{f179}'); // 
-        m.insert("gitignore_global", '\u{f1d3}'); // 
         m.insert("go.mod", '\u{e626}'); // 
         m.insert("go.sum", '\u{e626}'); // 
         m.insert("gradle", '\u{e256}'); // 
```
Comment (cafkafk) [inline src/output/icons.rs:72]: all from this point should be a separate PR imo
